x/ibc-hooks/contracts/crosschain-swaps/schema/crosschain-swaps.json

```diff
@@ -0,0 +1,270 @@
+                  },
+                  {
+                    "type": "null"
+                  }
+                ]
+              },
+              "input_coin": {
+                "$ref": "#/definitions/Coin"
+              },
+              "output_denom": {
+                "type": "string"
+              },
+              "receiver": {
+                "$ref": "#/definitions/Addr"
+              },
+              "slippage": {
+                "$ref": "#/definitions/Slippage"
+              }
+            },
+            "additionalProperties": false
+          }
+        },
+        "additionalProperties": false
+      },
+      {
+        "description": "Executing a recover will transfer any recoverable tokens that the sender has in this contract to its account.\n\nThis is only usable if the contract is configured with track_ibc_sends.\n\nThe only tokens that are considered recoverable for a \"sender\" are those returned by an IBC transfer sent by this contract, that are known to have failed, and that originated with a message specifying the \"sender\" as its recovery address.",
+        "type": "object",
+        "required": [
+          "recover"
+        ],
+        "properties": {
+          "recover": {
+            "type": "object",
+            "additionalProperties": false
+          }
+        },
+        "additionalProperties": false
+      }
+    ],
+    "definitions": {
+      "Addr": {
+        "description": "A human readable address.\n\nIn Cosmos, this is typically bech32 encoded. But for multi-chain smart contracts no assumptions should be made other than being UTF-8 encoded and of reasonable length.\n\nThis type represents a validated address. It can be created in the following ways 1. Use `Addr::unchecked(input)` 2. Use `let checked: Addr = deps.api.addr_validate(input)?` 3. Use `let checked: Addr = deps.api.addr_humanize(canonical_addr)?` 4. Deserialize from JSON. This must only be done from JSON that was validated before such as a contract's state. `Addr` must not be used in messages sent by the user because this would result in unvalidated instances.\n\nThis type is immutable. If you really need to mutate it (Really? Are you sure?), create a mutable copy using `let mut mutable = Addr::to_string()` and operate on that `String` instance.",
+        "type": "string"
+      },
+      "Coin": {
+        "type": "object",
+        "required": [
+          "amount",
+          "denom"
+        ],
+        "properties": {
+          "amount": {
+            "$ref": "#/definitions/Uint128"
+          },
+          "denom": {
+            "type": "string"
+          }
+        }
+      },
+      "Decimal": {
+        "description": "A fixed-point decimal value with 18 fractional digits, i.e. Decimal(1_000_000_000_000_000_000) == 1.0\n\nThe greatest possible value that can be represented is 340282366920938463463.374607431768211455 (which is (2^128 - 1) / 10^18)",
+        "type": "string"
+      },
```

Comment:
wtf, this typedef isn't just imported from somewhere :/

Reply:
🤷 I'm sure that's one of the many opinions Dan has about json schemas.